### PR TITLE
feat: hide empty stages

### PR DESCRIPTION
**Pull request by mQzLjP**
> only display health check section if it fails (list all issues as bullet points). if it passes, then don't show anything

This pull request is for this task, but for "list all issue as bullet points", I don't quite know if I need to do anything on my side.

### CODE REVIEW

**Comment by BoYanZh** [comment]
Can we just hide all results with 0 scores and empty comments? The stage with the healthcheck parser can have an arbitrary name.

File: joint_teapot/utils/joj3.py [hunk 1]
```diff
@@ -196,6 +196,11 @@ def generate_title_and_comment(
         + "[Joint-Teapot](https://github.com/BoYanZh/Joint-Teapot) with ❤️.\n"
     )
     for stage in stages:
+        stage_score = 0
+        for i, result in enumerate(stage["results"]):
```
Comment (BoYanZh) [inline joint_teapot/utils/joj3.py:200]: `enumerate` is not necessary here

also, consider `stage_score = sum(result["score"] for result in stage["results"])`

File: joint_teapot/utils/joj3.py [hunk 2]
```diff
@@ -196,6 +196,11 @@ def generate_title_and_comment(
...
+            stage_score += result["score"]
+        if stage_score == 0 and stage["results"][0]["comment"].strip() == "":
```
Comment (BoYanZh) [inline joint_teapot/utils/joj3.py:202]: We need to make sure every comment in this stage is empty.

`all(result["comment"].strip() == "" for result in stage["results"])`

**Comment by mQzLjP** [comment]
Done @BoYanZh

**Review by BoYanZh — CHANGES_REQUESTED** [review]

File: joint_teapot/utils/joj3.py (hunk 1 above)
Comment (BoYanZh) [reply joint_teapot/utils/joj3.py:200]: better to check all scores are 0 to avoid a coincidence sum of 0.

**Pull request merged**